packages/story-editor/src/components/library/panes/shopping/productDropdown.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import { useFeature } from 'flagged';
+
+/**
+ * External dependencies
+ */
+import { __ } from '@googleforcreators/i18n';
+import { useCallback, useEffect, useState } from '@googleforcreators/react';
+import { Datalist } from '@googleforcreators/design-system';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { useStory, useAPI } from '../../../../app';
+
+const StyledDropDown = styled(Datalist.DropDown)`
+  width: 240px;
+`;
+function ProductDropdown({ product, setProduct }) {
+  const isEnabled = useFeature('shoppingIntegration');
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [initialOptions, setInitialOptions] = useState([]);
+
+  const {
+    actions: { getProducts },
+  } = useAPI();
+  const isSaving = useStory(
+    ({
+      state: {
+        meta: { isSaving },
+      },
+    }) => isSaving
+  );
+
+  useEffect(() => {
+    (async () => {
+      setIsLoading(true);
+      const products = await getProducts();
+      setInitialOptions(
+        products.map((p) => ({
+          name: p.productTitle,
+          id: p.productId,
+          product: p,
+        }))
```

Comment:
This chunk of code is a repeat of logic in `getProductsByQuery`. Is there a chance to re-use this, even if it means submitting an empty string to `getProducts`

Reply:
Updated here:
https://github.com/GoogleForCreators/web-stories-wp/pull/11359/files#diff-df9e7b1e3b42a1f29c8c10f8ecd4a1d1a7d1c80201a9d31f2c8f91bf02911683R57